### ISSUE TITLE
"Picking up the rest" does not work anymore with fixed latest development repo

### ISSUE THREAD

**Issue by Gordon-Dry** [Gregory365/TPF2-Timetables#11]
It was a behaviour that a vehicle picked up the rest that was waiting at the platform after waiting by timetable.
This does not happen anymore.

**Comment by Gregory365** [comment]
This was intentional. 
Previously, no passengers or freight was picked up whilst a train was "stopped".
Now, passengers and freight should now be picked up as the train waits, so the extra wait at the end when it is time to depart is no longer necessary.

If you want a train to wait longer in a station to pick everything up, either change the timetable to have the train wait for longer or add a minimum wait time to the station in the line manager.

**Comment by Gordon-Dry** [comment]
There was a train (1 of 3) in a low rate long haul line with a beat of 45 minutes, as the setup of that line was not too long ago the 2nd train came too early and had to wait over 12 minutes and then departed without picking up the rest.

**Comment by Gordon-Dry** [comment]
No. As soon as the train reached the station it started picking up what was there.
Then it just waited for its turn, meanwhile the platform got filled with goods again.
Those have not been picked up as in earlier releases of the mod.

As the rate was well balanced only 8 of 176 items were left on the station - by two trains in a row exactly the same amount!
And as the trains left the station about 30 items were on the platform again.

**Comment by Gregory365** [comment]
I can't replicate this. For me, trains load all items whilst waiting at the station. Are you sure the items left on the station are for the line the train is on and not for a different line? 

I think the only way trains leave goods at the station is if the goods arrive just as the train is about to leave, in which case the train doesn't have time to pick all of the goods up before it departs.

**Comment by Gordon-Dry** [comment]
1. The cargo items which are at the station when the train arrives are all picked up, but what is stocked up while the train is waiting before departure is not picked after prior departure. This was different in earlier versions of the mod. The train just picked up more prior departure after the waiting phase.
2. Of course I am sure, there is only one line touching that station to pick up one specific good.
3. I assume this means the waiting delay done by the timetables mod is like a break between after loading and before departing and no additional pickup is triggered as in earlier versions of the mod.

**Comment by Gregory365** [comment]
1. Vehicles do now pick up cargo/passenger **during** the waiting phase. As a result, the picking up of goods after the departure time is no longer necessary. I made this change because it works easier with the current API and it allows for more reliable departure times for those that want to do slow/fast (local/express) trains.
I do understand that in some circumstances trains may leave before picking up all available goods but I think there are enough other mechanics to control departure adequately. For instance, you could:
- Change the departure time to be later
- Add a minimum wait time to the station
- ~~Set the train to only depart when it is fully loaded~~ (not currently working, let me know if this would be a solution though)

2. Great!

3. There isn't a break or time delay. 
In the previous version, the mod stops and then at departure time un-stops the train. When the train is un-stopped, the default behavior of the game is used (either wait for all goods on platform to board train OR wait for train to be full). 
In the current version, the mod tells the game to not depart the train and then at departure time forces the train to depart the station.

If you think there is a feature missing from this mod, then you're welcome to reply with what added behavior/mechanic you want, and why the current mechanics don't allow you to do what you want.

**Comment by Gordon-Dry** [comment]
I can try to set up a minimum wait time and the timetable set up.
But imho that would be counter-intuitive as I wanted the timetable to manage the job.
And as this line got a low rate a manual set up "minimum wait time" would have to be relatively high and could worsen the rate.
When does the counter for the "minimum wait time" begin? After reaching the station? When all available items are loaded and the train is not yet full?

**Comment by Gregory365** [comment]
The minimum wait time begins from when the train first arrives in the station.
How long does it take for your train to fill up? Surely setting a minimum wait time of 1 or 2 minutes would give the train plenty time to unload and reload without ruining the timetable?

I'm still not sure I understand why the train is leaving early for you. Are goods arriving to the station immediately just before the departure time? Is the train not loading whilst waiting before the departure time? Is the train very very long so loading takes a long time? 

In the scenarios I've been testing I haven't experienced this issue at all so I'm struggling to understand the situation.

**Comment by Gordon-Dry** [comment]
The above mentioned scenario was like that:
1. Goods are coming in all the time, but slowly
2. When the train arrives, it loads 168 items, 8 empty slots (capacity is 176)
3. It waits for its scheduled departure which is several minutes later because it was early
4. Meanwhile about 30 items are on the station platform
5. The train departs without loading them
6. The vanilla settings are "load when available", "minimum wait time 0", "maximum wait time 3 minutes", so all default values

It looks like this for me:
- The train "sees 168 items on the platform" as soon as stopped and ready to load
- It loads them and then proceeds
- Usually this would be "depart now", with the timetables mod it's "wait for scheduled departure"

I was updating some mods and try new Tpf2 beta build 35300 now, I will test the behaviour by increasing the minimum wait time.

**Comment by Gregory365** [comment]
Ok thanks for the reply, so the bug here is that loading isn't occurring whilst the train is waiting. 

I don't know why that is happening but I'll have another try to see if I can replicate.
